### PR TITLE
Fix the generate mutation and navigate on success

### PR DESCRIPTION
**Pull request by Adamk8**
The mutation should now work, if it receives a success message it will navigate to the schedule page

### CODE REVIEW

File: src/pages/Generate.tsx [hunk 1]
```diff
@@ -8,29 +8,64 @@ import {
...
+
+  useEffect(() => {
+    if (!loading) {
+      if (data && !error) {
+        if (data.generateSchedule.success) {
+          navigate("/schedule");
+        } else {
+          console.log(data);
+          toast({
+            title: "Failed to Generate Scheudle",
```
Comment (jayden-chan) [inline src/pages/Generate.tsx]:
```suggestion
            title: "Failed to Generate Schedule",
```

File: src/pages/Generate.tsx [hunk 2]
```diff
@@ -8,29 +8,64 @@ import {
...
+            description: data.generateSchedule.message,
+            status: "error",
+            isClosable: true,
+          });
+        }
+      } else if (error) {
+        console.log(error);
+        toast({
+          title: "Failed to Generate Scheudle",
```
Comment (jayden-chan) [inline src/pages/Generate.tsx]:
```suggestion
          title: "Failed to Generate Schedule",
```

File: src/pages/Generate.tsx [hunk 3]
```diff
@@ -8,29 +8,64 @@ import {
...
   const [term, setTerm] = useState("");
 
   const bg = useColorModeValue("gray.50", "gray.700");
 
+  const toast = useToast();
+
   const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    submit({ variables: { year, term } });
     e.preventDefault();
+    let input = {
```
Comment (jayden-chan) [inline src/pages/Generate.tsx]:
```suggestion
    const input = {
```

**Review by jayden-chan — APPROVED** [review]
Looks good! Just a couple optional minor errors

**Review by cbuchko — APPROVED** [review]

**Pull request merged**